### PR TITLE
refac(log): use tracing

### LINKED ISSUE #10: refac(log): use crate tracing instead of fern
we introduce tokio in one of the PR, should be nice to use https://github.com/tokio-rs/tracing now

### PR DESCRIPTION
**Pull request by edouardparis**
close #10

### CODE REVIEW

File: src/revaultd/mod.rs [hunk 1]
```diff
@@ -42,6 +43,9 @@ pub struct RevaultD {
 
 impl RevaultD {
     pub fn new(config: &Config) -> Result<RevaultD, RevaultDError> {
+        let span = span!(Level::INFO, "revaultd");
+        let _enter = span.enter();
```
Comment (darosior) [inline src/revaultd/mod.rs:47]: nit: why keeping it if you don't use it ?

File: src/revaultd/mod.rs [hunk 2]
```diff
@@ -70,11 +74,14 @@ impl RevaultD {
         method: &str,
         input: Option<T>,
     ) -> Result<U, RevaultDError> {
+        let span = span!(Level::INFO, "request");
+        let _guard = span.enter();
```
Comment (darosior) [inline src/revaultd/mod.rs:78]: same

**Review by darosior — APPROVED** [review]
ACK 0e56ca0df0e5e7261c4bcf8adef544c462a772c5

File: src/revaultd/mod.rs (hunk 1 above)
Comment (edouardparis) [reply src/revaultd/mod.rs:47]: you need to allocate it and then the span will be closed when the var is dropped

Comment (darosior) [reply src/revaultd/mod.rs:47]: Ok, would have thought the compiler would figure out the scope by itself

**Pull request merged**